src/ApiException.php

```diff
@@ -138,9 +181,41 @@ public static function createFromRestApiResponse(
             $previous
         );
     }
+    
+    /**
+     * Checks if decoded metadata includes errorInfo message.
+     * @param array $decodedMetadata
+     * @return array $errorInfo {
+     *     @type boolean $containsErrorInfo
+     *     @type string $reason
+     *     @type string $domain
+     *     @type array $errorInfoMetadata
+     * }
+     */
+    public static function containsErrorInfo(array $decodedMetadata)
```

Comment:
I think this also should _probably_ be private. I see no reason to expose this to the end user
```suggestion
    private static function containsErrorInfo(array $decodedMetadata)
```
```suggestion
    public static function containsErrorInfo(array $decodedMetadata)
```